config.w32

```diff
@@ -235,17 +235,17 @@ if (PHP_MONGODB != "no") {
   if (PHP_MONGODB_SASL != "no" &&
       CHECK_LIB("libsasl.lib", "mongodb", PHP_MONGODB) &&
       CHECK_HEADER_ADD_INCLUDE("sasl/sasl.h", "CFLAGS_MONGODB")) {
-    mongoc_opts.MONGOC_ENABLE_SASL = 1;
 
     // TODO 3.0: Remove warning on "yes" as it implies "sspi"
     if (PHP_MONGODB_SASL == "yes") {
-      WARNING("Cyrus SASL support for Windows was removed. Falling back to SSPI.");
+      WARNING("Cyrus SASL support for Windows was removed. Falling back to SSPI; use '--with-mongodb-sasl=sspi' to avoid this warning.");
     }
 
     if (PHP_MONGODB_SASL == "yes" || PHP_MONGODB_SASL == "sspi") {
+      mongoc_opts.MONGOC_ENABLE_SASL = 1;
       mongoc_opts.MONGOC_ENABLE_SASL_SSPI = 1;
     } else {
-      WARNING("mongodb sasl support not enabled, unknown value for PHP_MONGODB_SASL: " + PHP_MONGODB_SASL);
+      WARNING("MongoDB SASL support not enabled, unknown value for PHP_MONGODB_SASL: " + PHP_MONGODB_SASL);
     }
 
     if (CHECK_FUNC_IN_HEADER("sasl/sasl.h", "sasl_client_done")) {
```

Comment:
If the check above is updated, suggest also updating the warning message below (when `PHP_MONGODB_SASL != "no"`):

```sh
WARNING("MongoDB SASL support not enabled");
```

Consider changing to a non-WARNING message since I expect this would only be printed if a user chose `--with-mongodb-sasl=no`.

Reply:
Agree. With the removed checks for system libraries, this condition will no longer apply. I did change the warning on an unknown value for `--with-mongodb-sasl` to an error as we shouldn't continue in that case.